complete/src/main/java/hello/FileUploadController.java

```diff
@@ -45,15 +45,15 @@ public String handleFileUpload(@RequestParam("name") String name,
 			redirectAttributes.addFlashAttribute("message", "Folder separators not allowed");
 			return "redirect:/";
 		}
-		if (name.contains("/")) {
+		if (name.contains("\\")) {
 			redirectAttributes.addFlashAttribute("message", "Relative pathnames not allowed");
```

Comment:
Well, "/" IS a relative path separator for Unix/Mac systems. Point taken, this isn't platform friendly enough.